invenio_vocabularies/contrib/names/datastreams.py

```diff
@@ -43,17 +44,18 @@ def __init__(self, origin=None, mode="r", since=None, *args, **kwargs):
         self.s3_client = S3OrcidClient()
         self.since = since
 
-    def _fetch_orcid_data(self, orcid_to_sync, bucket):
+    def _fetch_orcid_data(self, flask_app, orcid_to_sync, bucket):
         """Fetches a single ORCiD record from S3."""
         # The ORCiD file key is located in a folder which name corresponds to the last three digits of the ORCiD
         suffix = orcid_to_sync[-3:]
         key = f"{suffix}/{orcid_to_sync}.xml"
+        flask_app.logger.debug(f"Fetching ORCiD record: {key} from bucket: {bucket}")
         try:
             # Potential improvement: use the a XML jax parser to avoid loading the whole file in memory
             # and choose the sections we need to read (probably the summary)
             return self.s3_client.read_file(f"s3://{bucket}/{key}")
         except Exception:
-            current_app.logger.exception("Failed to fetch ORCiD record.")
+            flask_app.logger.exception(f"Failed to fetch ORCiD record: {key}")
```

Comment:
in your other [PR](https://github.com/inveniosoftware/invenio-rdm-records/pull/1989) you are using current_app, could you explain why we need flask app here?

Reply:
This is being executed in a thread outside the app context, we need to pass it as an argument, I called it flask_app to avoid confusion but is the same as current_app

Reply:
in my thinking, normally invenio calls such variables only `app`. naming it `flask_app` could add more confusion than resolving confusion

Reply:
My idea was to differentiate it since it’s not the proxy we typically use, but the actual Flask app instance passed to the thread. That said, I’m happy to rename it to `app` if you think it’s clearer and more consistent.

Reply:
if we compare it with what we have on the `ext.py` files like [here](https://github.com/inveniosoftware/invenio-jobs/blob/master/invenio_jobs/ext.py#L40). the `app` there and the `flask_app` are the type wise the same?

Reply:
Right, they are the same. In `ext.py`, it’s clear we’re working with the Flask app since it’s being instantiated there. Here, it’s a bit less obvious that we’re dealing with a Flask-based app. In any case, I’ve updated the name for consistency.